server/src/test/scala/org/elasticmq/server/config/ElasticMQServerConfigTest.scala

```diff
@@ -32,13 +33,59 @@ class ElasticMQServerConfigTest extends AnyFunSuite with Matchers with OptionVal
 
   test("Normal queues should not have appended .fifo suffix") {
     val conf = new ElasticMQServerConfig(ConfigFactory.load("test"))
-    val normalQueues = conf.createQueues.filter(!_.isFifo)
+    val normalQueues = conf.readQueuesToLoad().filter(!_.isFifo)
     normalQueues.foreach(_.name should not endWith ".fifo")
   }
 
   test("FIFO queue should have appended .fifo suffix") {
     val conf = new ElasticMQServerConfig(ConfigFactory.load("test"))
-    val fifoQueue = conf.createQueues.find(_.isFifo).value
+    val fifoQueue = conf.readQueuesToLoad().find(_.isFifo).value
     fifoQueue.name shouldBe "fifoQueue.fifo"
   }
+
+  test("Should correctly parse persisted queues configuration") {
+    val conf = new ElasticMQServerConfig(ConfigFactory.load("test"))
+    val config = ConfigFactory.parseString(server.load(this.getClass, "expected-single-queue-config.conf"))
+    val persistedQueues = conf.readPersistedQueues(Some(config))
+    val expectedQueue = CreateQueue(
+      "test",
+      Some(3L),
+      Some(0L),
+      Some(0L),
+      Some(DeadLettersQueue("dead", 4)),
+      isFifo = false,
+      hasContentBasedDeduplication = true,
+      Some("copyTo"),
+      Some("messageTo"),
+      Map("tag1Key" -> "tag1Value")
+    )
+    persistedQueues.length shouldBe 1
+    persistedQueues.head shouldBe expectedQueue
+  }
+
+  test("Should not parse persisted queues when disabled") {
+    val conf = new ElasticMQServerConfig(ConfigFactory.load("test"))
+    val persistedQueues = conf.readPersistedQueues()
+    persistedQueues shouldBe Nil
+  }
+
+  test("Persisted queues should take precedence over startup queues with same names") {
+    val conf = new ElasticMQServerConfig(ConfigFactory.load("test-with-queue-storage-enabled.conf"))
+    val config = ConfigFactory.parseString(server.load(this.getClass, "two-queues-config.conf"))
+    val persistedQueues = conf.readQueuesToLoad(conf.readPersistedQueues(Some(config)))
+    val expectedQueue = CreateQueue(
+      "test-2",
+      Some(3L),
+      Some(0L),
+      Some(0L),
+      Some(DeadLettersQueue("dead-2", 4)),
+      isFifo = false,
+      hasContentBasedDeduplication = true,
+      Some("copyTo"),
+      Some("messageTo"),
+      Map("tag12Key" -> "tag12Value")
+    )
+    persistedQueues.length shouldBe 3
+    persistedQueues.head shouldBe expectedQueue
```

Comment:
this is the modified queue? shouldn't we check the other ones in the list as well?

Reply:
would be helpful to somehow write in the test which queues are from the conf, which from the persistence file, and how they are modified